### PR TITLE
Quicksearch: use class icons when available

### PR DESCRIPTION
**Pull request by NiklasBr**
## Changes in this pull request  
Show custom class icon in quicksearch for root products (variants still keep their icon)

## Additional info  

image

### CODE REVIEW

**Comment by NiklasBr** [comment]
These errors make no sense considering the changes:
```log
Error: ] Found 21 errors                                                        

Error: Instantiated class Google_Service_Customsearch not found.
Error: Access to property $cse on an unknown class Google_Service_Customsearch.
Error: Parameter $googleResponse of method Pimcore\Google\Cse::readGoogleResponse() has invalid typehint type Google_Service_Customsearch_Search.
Error: Parameter $googleResponse of method Pimcore\Google\Cse::readGoogleResponse() has invalid typehint type Google_Service_Customsearch_Search.
Error: Call to method getSearchInformation() on an unknown class Google_Service_Customsearch_Search.
Error: Call to method getItems() on an unknown class Google_Service_Customsearch_Search.
Error: PHPDoc tag @var for variable $results contains unknown class Google_Service_Customsearch_Result.
Error: Call to method getPagemap() on an unknown class Google_Service_Customsearch_Result.
Error: Access to offset 'pagemap' on an unknown class Google_Service_Customsearch_Result.
Error: Access to offset 'pagemap' on an unknown class Google_Service_Customsearch_Result.
Error: Access to offset 'pagemap' on an unknown class Google_Service_Customsearch_Result.
Error: Access to offset 'pagemap' on an unknown class Google_Service_Customsearch_Result.
Error: Access to offset 'pagemap' on an unknown class Google_Service_Customsearch_Result.
Error: Call to method getLink() on an unknown class Google_Service_Customsearch_Result.
Error: Property Pimcore\Google\Cse\Item::$raw has unknown class Google_Service_Customsearch_Result as its type.
Error: Parameter $data of method Pimcore\Google\Cse\Item::__construct() has invalid typehint type Google_Service_Customsearch_Result.
Error: Parameter $data of method Pimcore\Google\Cse\Item::__construct() has invalid typehint type Google_Service_Customsearch_Result.
Error: Parameter $data of method Pimcore\Google\Cse\Item::setValues() has invalid typehint type Google_Service_Customsearch_Result.
Error: Parameter $data of method Pimcore\Google\Cse\Item::setValues() has invalid typehint type Google_Service_Customsearch_Result.
Error: Parameter $raw of method Pimcore\Google\Cse\Item::setRaw() has invalid typehint type Google_Service_Customsearch_Result.
Error: Return typehint of method Pimcore\Google\Cse\Item::getRaw() has invalid type Google_Service_Customsearch_Result.
Error: Process completed with exit code 1.
```

**Comment by dvesh3** [comment]
@NiklasBr seems like they changed the namespace in `googleapis/google-api-php-client-services` on v0.2 so stan is unable to find the class. we are on it!

File: bundles/AdminBundle/Controller/Searchadmin/SearchController.php [hunk 1]
```diff
@@ -482,6 +482,14 @@ public function quicksearchAction(Request $request, EventDispatcherInterface $ev
 
                 if ($element instanceof Asset) {
                     $data['iconCls'] .= ' pimcore_icon_' . \Pimcore\File::getFileExtension($element->getFilename());
+                 } elseif ($element instanceof DataObject\Concrete) {
+                    // Show custom on base class (not variants) if it exists
+                    if ($element->getType() === DataObject::OBJECT_TYPE_OBJECT && $element->getClass()->getIcon()) {
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Searchadmin/SearchController.php:491]: I think we should use the same concept here as we do for the tree, see: https://github.com/pimcore/pimcore/blob/2f18a0f26380c56b609d937b40d2050c8d1a3e1a/bundles/AdminBundle/Controller/Traits/AdminStyleTrait.php#L37

**Review by brusch — CHANGES_REQUESTED** [review]

**Comment by NiklasBr** [comment]
Rebase:ing is the bane of my existence.

**Pull request closed without merging**